frontend/src/components/vue_canvas/instance_list.vue

```diff
@@ -269,11 +271,24 @@
 
 
           // MAIN loop
+          ctx.textAlign = "start";
+          ctx.setLineDash([0])
 
-          for (var i in this.instance_list) {
-            this.draw_single_instance(ctx, i)
+
+          // var startTime = performance.now();
+          if(!this.canvas_mouse_tools || this.video_mode){
+            for (let i in this.instance_list) {
```

Comment:
this is a "workaround" while we figure out best way to have the class structure with the visible thing

e.g. each instance has a `in_viewport` flag